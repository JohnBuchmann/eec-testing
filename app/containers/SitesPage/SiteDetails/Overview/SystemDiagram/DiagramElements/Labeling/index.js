import { makeStyles } from '@material-ui/core';
import React from 'react';
import PropTypes from 'prop-types';
import { Colors } from 'Theme';

/**
 * @property {Object} useStyles Stores the styles
 */
const useStyles = makeStyles({
  evTitle: {
    fontFamily: 'Inter, sans-serif',
    color: Colors.surfCrest,
    fontSize: '11px',
    fontWeight: '500',
    textTransform: 'uppercase',
  },
  label: {
    fontFamily: 'Inter, sans-serif',
    color: Colors.white,
    fontSize: '11px',
  },
  labelTitle: {
    fontSize: '12px',
    fontWeight: '600',
  },
});

const Labeling = (props) => {
  const classes = useStyles();
  const { x = 0, y = 0, label, description } = props;

  const dxPaddingValue = 5;

  return (
    <>
      <desc>Label</desc>
      <text x={x} y={y} className={classes.label}>
        <tspan className={classes.labelTitle}>{label}:</tspan>
        <tspan dx={dxPaddingValue} className={classes.totalLoadTitle}>
          {description}
        </tspan>
      </text>
    </>
  );
};

Labeling.propTypes = {
  x: PropTypes.number,
  y: PropTypes.number,
  label: PropTypes.string,
  description: PropTypes.string,
};

export default Labeling;
