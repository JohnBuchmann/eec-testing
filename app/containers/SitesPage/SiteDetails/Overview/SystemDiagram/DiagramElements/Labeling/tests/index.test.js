import { mount } from 'enzyme';
import React from 'react';
import Labeling from '..';

describe('Labeling', () => {
  it('should display label with default values', () => {
    const actualComponent = mount(
      <svg xmlns="http://www.w3.org/2000/svg">
        <Labeling />
      </svg>
    );

    expect(actualComponent).toMatchSnapshot();
  });

  it('should display label', () => {
    const positionX = 100;
    const positionY = 100;
    const label = 'Test';
    const description = 'Testing';

    const actualComponent = mount(
      <svg xmlns="http://www.w3.org/2000/svg">
        <Labeling
          x={positionX}
          y={positionY}
          label={label}
          description={description}
        />
      </svg>
    );

    expect(actualComponent).toMatchSnapshot();
  });
});
