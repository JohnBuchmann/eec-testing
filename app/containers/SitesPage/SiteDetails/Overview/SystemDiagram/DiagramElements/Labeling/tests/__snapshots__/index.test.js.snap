// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labeling should display label 1`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
>
  <Labeling
    description="Testing"
    label="Test"
    x={100}
    y={100}
  >
    <desc>
      Label
    </desc>
    <text
      className="makeStyles-label-2"
      x={100}
      y={100}
    >
      <tspan
        className="makeStyles-labelTitle-3"
      >
        Test
        :
      </tspan>
      <tspan
        dx={5}
      >
        Testing
      </tspan>
    </text>
  </Labeling>
</svg>
`;

exports[`Labeling should display label with default values 1`] = `
<svg
  xmlns="http://www.w3.org/2000/svg"
>
  <Labeling>
    <desc>
      Label
    </desc>
    <text
      className="makeStyles-label-2"
      x={0}
      y={0}
    >
      <tspan
        className="makeStyles-labelTitle-3"
      >
        :
      </tspan>
      <tspan
        dx={5}
      />
    </text>
  </Labeling>
</svg>
`;
